qiskit/optimization/algorithms/optimization_algorithm.py

```diff
@@ -217,11 +173,29 @@ def status(self) -> OptimizationResultStatus:
         """
         return self._status
 
-    @status.setter  # type: ignore
-    def status(self, status: OptimizationResultStatus) -> None:
-        """Set a new termination status.
+    @property
+    def variables(self) -> Optional[List[Variable]]:
+        """Returns the list of variables of the optimization problem.
 
-        Args:
-            status: The new termination status.
+        Returns:
+            The list of variables.
         """
-        self._status = status
+        return self._variables
+
+    @property
+    def variables_dict(self) -> Dict[str, int]:
+        """Returns the pairs of variable names and values under optimization.
```

Comment:
Would this be better saying 'returns the optimal value a dictionary of the variables name and corresponding value' Saying 'pairs' makes me think too much of Tuple.

Reply:
Agree. I change it.